codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/ProtocolGenerator.java

```diff
@@ -120,8 +121,7 @@ default ApplicationProtocol resolveApplicationProtocol(
      *
      * @param context Serde context.
      */
-    default void generateSharedComponents(GenerationContext context) {
-    }
+    void generateSharedSerializerComponents(GenerationContext context);
```

Comment:
Why the name change and removal of default?

Reply:
Should have kept the `default` attribute, name change is because the the shared components for serializer will be generated in a file separate file from shared deserializer componenets.